src/js/stores/AppsStore.js

```diff
@@ -10,6 +11,24 @@ function removeApp(apps, appId) {
   }).value();
 }
 
+function removeTask(tasks, relatedAppId, taskId) {
+  return lazy(tasks).reject(function (task) {
+    return task.id === taskId && task.appId === relatedAppId;
+  }).value();
+}
+
+function processCurrentApp(currentApp) {
```

Comment:
Without the "only" your sentence is correct.
(currentApp == null) should be an edge case (strange response from server).
Usually this should be filled with the app data requested from the server.